### PR TITLE
implementing the LMOVE and BLMOVE commands

### LINKED ISSUE #1447: BRPOPLPUSH deprecated from Redis 6.2.0.
It seems that BRPOPLPUSH command is considered deprecated from Redis version 6.2.0 and will be replaced by BLMOVE.

- https://redis.io/commands/brpoplpush
- https://redis.io/commands/blmove

Best regards,
Cristiano Urban.

### LINKED ISSUE #1503: redis lmove
Hello everybody
I have seen that redis-py library is no longer updated since June of 2020, although since then it was regularly updated on a quarterly basis. Considering this, do you perhaps have any idea of when the next release of that library, including the LMove method, will be distributed?

### PR DESCRIPTION
**Pull request by chayim**
closes #1503
closes #1447 

Note - this passes in completion again - once [my pr to fix unit tests](https://github.com/andymccurdy/redis-py/pull/1499) is merged in. Checkboxes here are checked based on that assumption.

### Pull Request check-list

_Please make sure to review and check all of these items:_

- [X] Does `$ tox` pass with this change (including linting)?
- [X] Do the CI tests pass with this change (enable it first in your forked repo and wait for the github action build to finish)?
- [X] Is the new or changed code fully tested?
- [X] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)?

_NOTE: these things are not required to open a PR and can be done
afterwards / while the PR is open._

### Description of change

Added support for the redis [lmove](https://redis.io/commands/lmove) and [blmove](https://redis.io/commands/blmove) command, new in 6.2. See #1434

### CODE REVIEW

File: redis/client.py [hunk 1]
```diff
@@ -561,7 +561,7 @@ class Redis:
     """
     RESPONSE_CALLBACKS = {
         **string_keys_to_dict(
-            'AUTH EXPIRE EXPIREAT HEXISTS HMSET MOVE MSETNX PERSIST '
+            'AUTH EXPIRE EXPIREAT HEXISTS HMSET LOVE MOVE MSETNX PERSIST '
```
Comment (abrookins) [inline redis/client.py:564]: Is this supposed to be LMOVE? Should we also have BLMOVE here?

Comment (chayim) [reply redis/client.py:564]: Yes and updated. More interesting that it didn't fail the tests. Probably should run vulture against these.

File: redis/client.py [hunk 2]
```diff
@@ -1822,6 +1822,21 @@ def keys(self, pattern='*'):
...
+        Returns the element being popped and pushed.
+        """
+        params = [first_list, second_list, src, dest]
+        return self.execute_command("LMOVE", *params)
+
+    def blmove(self, first_list, second_list, src="LEFT", dest="RIGHT"):
+        """
+        Blocking version of lmove.
+        """
+        return self.lmove(first_list, second_list, src, dest)
```
Comment (abrookins) [inline redis/client.py:1838]: Why are you calling lmove here instead of executing blmove? Maybe I don't understand something!

**Comment by chayim** [comment]
@abrookins The recent [doc update for blmove](https://redis.io/commands/blmove) now shows that a timeout was required. Updates to blmove and tests to reflect that are now in this PR.

**Pull request merged**